lib/schedule.js

```diff
@@ -14,31 +14,26 @@ var events = require('events'),
 /* Job object */
 var anonJobCounter = 0;
 
-function Job() {
-  var name;
-
-  // process arguments to the constructor
-  var arg;
-  for (var i = 0; i < arguments.length; i++) {
-    arg = arguments[i];
-    if (typeof arg == 'string' || arg instanceof String) {
-      name = arg;
-    } else {
-      this.job = arg;
-    }
-  }
-
-  // give us a random name if one wasn't provided
-  if (name == null) {
-    name = '<Anonymous Job ' + (++anonJobCounter) + '>';
-  }
-
+function Job(name, job, callback) {
   // setup a private pendingInvocations variable
   var pendingInvocations = [];
 
+  // Set scope vars
+  var jobName = name && typeof name === 'string' ? name : '<Anonymous Job ' + (++anonJobCounter) + '>';
+  this.job = name && typeof name === 'function' ? name : job;
+
+  // Make sure callback is actually a callback
+  if (this.job === name) {
+    // Name wasn't provided and maybe a callback is there
+    this.callback = job || false;
+  } else {
+    // Name was provided, and maybe a callback is there
+    this.callback = callback || false;
+  }
+
```

Comment:
May you could check here if callback it's a function or not instead of in prepareNextInvocation